gadget/gadget.go

```diff
@@ -1298,12 +1298,20 @@ var disallowedKernelArguments = []string{
 	"init",
 }
 
+var allowedSnapdKernelArguments = []string{
+	"snapd_system_disk",
```

Comment:
Should we use `snapd.system_disk` for consistency with the other option?

Reply:
which other option?

Reply:
We already have `snapd.debug` below

Reply:
I think snapd.debug is the historically strange one, also that one is really conceptually mainly about controlling the snapd daemon from the kernel command line.  I think snapd_system_disk is more related to snapd_recovery_mode etc.

Reply:
Kernel will log a notice message during initialization: `Unknown kernel command line parameters "%s", will be passed to user space.`. But only for names that do not contain "`.`", because those with a dot are supposed to be for modules, not for initialization. Systemd uses dots. It is not that important. But there is this slight difference of behavior.

Reply:
Also, according to [1] if the kernel doesn't recognize a parameter and it doesn’t contain a ‘.’, the parameter gets passed to init (as an argument or as an env var), which we should try to avoid. systemd is probably just ignoring what it does not understand, but anyway...

[1] https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html

Reply:
systemd ignores [unknown options](https://github.com/systemd/systemd/blob/7ef09e2099a4f97ad40748d6b7c735b45aa4c990/src/core/main.c#L1083) and [all positional arguments](https://github.com/systemd/systemd/blob/7ef09e2099a4f97ad40748d6b7c735b45aa4c990/src/core/main.c#L1096) when PID is 1.

I suppose it needs to do it arguments like `splash` anyway. Positional arguments are always ignored. But we should never have parameters starting with a dash (`-`), in case it becomes a valid systemd parameter.